cmake/external/ObsPluginHelpers.cmake

```diff
@@ -1,163 +1,348 @@
-# Functions for generating external plugins
+if(POLICY CMP0087)
+  cmake_policy(SET CMP0087 NEW)
+endif()
 
-set(EXTERNAL_PLUGIN_OUTPUT_DIR "${CMAKE_BINARY_DIR}/rundir")
+set(OBS_STANDALONE_PLUGIN_DIR "${CMAKE_SOURCE_DIR}/release")
+set(INCLUDED_LIBOBS_CMAKE_MODULES ON)
 
-# Fix XCode includes to ignore warnings on system includes
-function(target_include_directories_system _target)
+include(GNUInstallDirs)
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+  set(OS_MACOS ON)
+  set(OS_POSIX ON)
+elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Linux|FreeBSD|OpenBSD")
+  set(OS_POSIX ON)
+  string(TOUPPER "${CMAKE_SYSTEM_NAME}" _SYSTEM_NAME_U)
+  set(OS_${_SYSTEM_NAME_U} ON)
+elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+  set(OS_WINDOWS ON)
+  set(OS_POSIX OFF)
+endif()
+
+# Old-Style plugin detected, find "modern" libobs variant instead and set global
+# include directories to fix "bad" plugin behaviour
```

Comment:
I'd prefer to avoid British English spellings in general, if possible.  "behavior"

Reply:
Oops - muscle memory... Fixed.